File: Samu/Pr04/src/Permiso.java
import java.util.*;


public class Permiso implements Comparable<Permiso> {
    private static String BD_SERVER = "localhost";
    private static String BD_NAME = "GI1819";

    private String rolName;
    private String pantalla;
    private boolean acceso;
    private boolean modificacion;


    public static List<Permiso> ListaPermisosRol(String rolName) {
        // Retorna una lista con todos los permisos de un rol determinado
        List<Permiso> lista = new ArrayList<Permiso>();
        BD miBD = new BD(BD_SERVER, BD_NAME);

        for (Object[] tupla : miBD.Select("SELECT * FROM tPermiso;")) {
            Permiso p = new Permiso((String) tupla[0], (String) tupla[1]);
            lista.add(p);
        }
        return lista;
    }

    public Permiso(String r, String p) {
        // Crea el objeto cargando sus valores de la base de datos
        BD miBD = new BD(BD_SERVER, BD_NAME);
        Object[] tupla = miBD.Select(String.format("select * from tPermiso where rolName = '%s' and pantalla = '%s';",
                r, p)).get(0);

        rolName = (String) tupla[0];
        pantalla = (String) tupla[1];
        acceso = (boolean) tupla[2];
        modificacion = (boolean) tupla[3];

    }

    public Permiso(String r, String p, boolean a, boolean m) {
        // Crea el objeto y lo inserta en la base de datos
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Insert(String.format("Insert into tPermiso(rolName, pantalla, acceso, modificacion) " +
                        "values ('%s', '%s', %s, %s);",
                r, p, a?1:0, m?1:0));
        this.rolName = r;
        this.pantalla = p;
        this.acceso = a;
        this.modificacion = m;

    }

    public void setRolName(String value) {
        // Actualiza el atributo en memoria y en la base de datos
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPermiso set rolName = '%s' where rolName = '%s';", value, this.rolName));
        this.rolName = value;

    }

    public String getRolName() {
        return rolName;
    }

    public String getPantalla() {
        return pantalla;
    }

    public void setPantalla(String value) {
        // Actualiza el atributo en memoria y en la base de datos
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPermiso set acceso = '%s' where pantalla = '%s';", value, this.pantalla));
        this.pantalla = value;

    }


    public boolean getAcceso() {
        return acceso;
    }

    public void setAcceso(boolean value) {
        // Actualiza el atributo en memoria y en la base de datos
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPermiso set acceso = %s where rolName = '%s' and pantalla = '%s';",
                value?1:0, this.rolName, this.pantalla));
        this.acceso = value;

    }


    public boolean getModificacion() {
        return modificacion;
    }

    public void setModificacion(boolean value) {
        // Actualiza el atributo en memoria y en la base de datos
        BD miBD = new BD(BD_SERVER, BD_NAME);
        miBD.Update(String.format("update tPermiso set modificacion = %s where rolName = '%s' and pantalla = '%s';",
                value?1:0, this.rolName, this.pantalla));
        this.modificacion = value;
    }

    @Override
    public boolean equals(Object o) {
        boolean res = false;
        if (o instanceof Permiso) {
            Permiso p = (Permiso) o;
            res = this.pantalla.equalsIgnoreCase(p.pantalla)
                    && this.rolName.equalsIgnoreCase(p.rolName);
        }
        return res;
    }

    public int hashCode() {
        return pantalla.toLowerCase().hashCode()
                + rolName.toLowerCase().hashCode();
    }

    public int compareTo(Permiso p) {
        int res = this.rolName.compareToIgnoreCase(p.rolName);
        if (res == 0) res = this.pantalla.compareToIgnoreCase(p.pantalla);
        return res;
    }
}

File: Aleks/HIS/Permiso.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using BDLibrary;

namespace HIS
{
    public class Permiso
    {

        private static string BD_SERVER = Properties.Settings.Default.BD_SERVER;
        private static string BD_NAME = Properties.Settings.Default.BD_NAME;

        private string rolName;
        private string pantalla;
        private bool acceso;
        private bool modificacion;


        public static List<Permiso> ListaPermisosRol(string rolName)
        {
		    List<Permiso> lista = new List<Permiso>();
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);

            foreach (object[] tupla in miBD.Select("SELECT pantalla FROM tPermiso WHERE rolName = '" + rolName + "';"))
            {
                string pantalla = (string)tupla[0];
                lista.Add(new Permiso(rolName, pantalla));
            }
		    return lista;
        }

        public Permiso(String r, String p)
        {
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
            object[] tupla = miBD.Select("SELECT * FROM tPermiso WHERE rolName = '" 
                            + r + "' AND pantalla = '" + p + "';")[0];
            rolName = (string)tupla[0];
            pantalla = (string)tupla[1];
            acceso = (bool)tupla[2];
            modificacion = (bool)tupla[3];
        }

        public Permiso(string r, string p, bool a, bool m)
        {
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);

            miBD.Insert("INSERT INTO tPermiso VALUES('" + r + "', '"
                + p + "', " + (a ? 1 : 0) + ", " + (m ? 1 : 0) + ");");

            rolName = r;
            pantalla = p;
            acceso = a;
            modificacion = m;
        }

        public void BorrarPermiso()
        {
            SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
            miBD.Update("DELETE tPermiso"
                    + "  WHERE rolName = '" + rolName
                    + "' AND pantalla = '" + pantalla + "';");
            rolName = pantalla = null;
            acceso = modificacion = false;
        }

        public string RolName
        {
            get { return rolName; }
            set
            {
                SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                miBD.Update("UPDATE tPermiso SET rolName = '" + value 
                        + "' WHERE rolName = '" + rolName 
                        + "' AND pantalla = '" + pantalla + "';");
                rolName = value;
            }
        }

        public string Pantalla
        {
            get { return pantalla; }
            set
            {
                SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                miBD.Update("UPDATE tPermiso SET pantalla = '" + value
                        + "' WHERE rolName = '" + rolName
                        + "' AND pantalla = '" + pantalla + "';");
                pantalla = value;
            }
        }

        public bool Acceso
        {
            get { return acceso; }
            set {
                    SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                    miBD.Update("UPDATE tPermiso SET acceso = " + (value?1:0)
                            + " WHERE rolName = '" + rolName
                            + "' AND pantalla = '" + pantalla + "';");
                    acceso = value; 
                }
        }

        public bool Modificacion
        {
            get { return modificacion; }
            set {
                    SQLSERVERDB miBD = new SQLSERVERDB(BD_SERVER, BD_NAME);
                    miBD.Update("UPDATE tPermiso SET modificacion = " + (value ? 1 : 0)
                            + " WHERE rolName = '" + rolName
                            + "' AND pantalla = '" + pantalla + "';");
                    modificacion = value; 
                }
        }

    }
}
